Prevent JS Issue if `woocommerce_available_variation` filter return false

EmranAhmed opened the pull request:
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Re-indexing `get_available_variations` method return values.

Suppose you made changed with `woocommerce_available_variation` hook and skip returning some specific variation product with false value. 

There is `array_filter` function to filter empty values from `get_available_variations` method but it preserves array index which convert to js object rather then js array on `single-product/add-to-cart/variable.php` template variation form `data-product_variations` attribute, and select box dropdown doesn't show. 

Just added `array_values` PHP function to re-index after `array_filter` function which convert to js array rather then js object before `get_available_variations` function return.

### How to test the changes in this Pull Request:

1. Suppose you want to hide only out of stock variation product. But you don't want to do it from **Settings -> Inventory -> Out of stock visibility**, Because Out of stock Single product also hide from catalog also. So you wrote this code to hide it.

```
add_filter( 'woocommerce_available_variation', function ( $variation, $productObject, $variationObject ) {
    return $variationObject->is_in_stock() ? $variation : FALSE;
}, 10, 3 );
```
It's fine that `get_available_variations` method use `array_filter` function to filter empty array values but it keeps index, thats why on `single-product/add-to-cart/variable.php` `available_variations` variable generate js object rather then js array and Variable drop down doesn't show.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

EmranAhmed committed:
Prevent JS Issue if `woocommerce_available_variation` return false for a variation.

After checking some condition if `woocommerce_available_variation` filter return false. Then Option list don't show because of JS issue.

kloon reviewed (changes requested):
Can you please update the PR by filling in the original template as requested. This PR is lacking details as to what you are trying to fix, steps to test as well as the build failing due to WP Coding Standards.

Please update the PR with the correct details and fix the PHPCS errors as per the build https://travis-ci.org/woocommerce/woocommerce/jobs/405961585 Or run PHPCS locally to find the errors.

EmranAhmed committed:
Re-indexing `get_available_variations` method values when `woocommerce_available_variation` filter return false.

### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

Re-indexing `get_available_variations` method values.

Suppose you made changed with `woocommerce_available_variation` hook and skip returning some specific variation product with false value. There is `array_filter` function to filter empty values from `get_available_variations` but it keeps index which convert to js object rather then array on `single-product/add-to-cart/variable.php` template variation form `data-product_variations` attribute, and select box dropdown doesn't show. I just added `array_values` PHP function to reindex after `array_filter` function which convert to js array rather then js object before `get_available_variations` function return.

Closes # .

### How to test the changes in this Pull Request:

1. Before approving pull req just return false on a specific variation via `woocommerce_available_variation` filter.
2. Now check that variation product dropdown doesn't show. Because it needs to reindex array.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

### Changelog entry

> Enter a short summary of all changes on this Pull Request. This will appear in the changelog if accepted.

kloon reviewed (approved):
Looks good, thanks for updating and fixing issues so the tests pass.
